### PR TITLE
MalwareBazaar Feed

### PR DESCRIPTION
**Pull request by bziser**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/35392
https://github.com/demisto/content-test-conf/pull/1337

## Description
Feed to MalwareBazaar

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [X] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (bakatzir) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:1]: revert

File: Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.yml [hunk 2]
```diff
@@ -0,0 +1,128 @@
+category: Data Enrichment & Threat Intelligence
+commonfields:
+  id: MalwareBazaar Feed
+  version: -1
+configuration:
+- display: Malware Bazaar API URL
```
Comment (bakatzir) [inline Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.yml:6]: no need to state malware bazzar again.
```suggestion
- display: Server URL
```

File: Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.yml [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  name: feedBypassExclusionList
+  required: false
+  type: 8
+- additionalinfo: Supports CSV values.
+  display: Tags
+  hidden: false
+  name: feedTags
+  required: false
+  type: 0
+- defaultvalue: 'false'
```
Comment (bakatzir) [inline Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.yml:97]: shouldn't it beset to true?
```suggestion
- defaultvalue: 'true'
```

File: Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed_description.md [hunk 4]
```diff
@@ -0,0 +1 @@
+Fetch indicators from a MalwareBazzaar feed. You can apply filters to the feed to focus which indicators you want to ingest.
```
Comment (bakatzir) [inline Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed_description.md:1]: usually used to have info regarding the authentication or different fetch filters. please add this info instead with examples, as this is mainly used for setup purposes, and not general info.

File: Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed_test.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
+"""Base Integration for Cortex XSOAR - Unit Tests file
```
Comment (bakatzir) [inline Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed_test.py:1]: rm this file

File: Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/README.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+This README contains the full documentation for your integration.
```
Comment (bakatzir) [inline Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/README.md:1]: please generate the readme using the demisto-sdk.

File: Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/test_data/baseintegration-dummy.json [hunk 7]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (bakatzir) [inline Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/test_data/baseintegration-dummy.json:1]: rm

File: Packs/MalwareBazaar/pack_metadata.json [hunk 8]
```diff
@@ -0,0 +1,15 @@
+{
+    "name": "MalwareBazaar Feed",
+    "description": "TODO",
```
Comment (bakatzir) [inline Packs/MalwareBazaar/pack_metadata.json:3]: TODO :)

File: Packs/MalwareBazaar/pack_metadata.json [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+    "support": "xsoar",
+    "currentVersion": "1.0.0",
+    "author": "Cortex XSOAR",
+    "url": "https://www.paloaltonetworks.com/cortex",
+    "email": "",
+    "categories": [
+        "Data Enrichment & Threat Intelligence"
+    ],
+    "tags": [],
```
Comment (bakatzir) [inline Packs/MalwareBazaar/pack_metadata.json:12]:
```suggestion
    "tags": [
        "Relationship"
    ],
```

File: Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.py [hunk 10]
```diff
@@ -0,0 +1,59 @@
+from typing import Dict
+from CommonServerPython import *  # noqa # pylint: disable=unused-wildcard-import
+from CommonServerUserPython import *  # noqa
+from JSONFeedApiModule import *
+
+INTEGRATION_NAME = "MalwareBazaar Feed"
+
+
+def custom_build_relationships(feed_config: Dict, mapping: Dict, indicator_data: Dict):
```
Comment (bakatzir) [inline Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.py:9]: add return type

File: Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.py [hunk 11]
```diff
@@ -0,0 +1,59 @@
+from typing import Dict
+from CommonServerPython import *  # noqa # pylint: disable=unused-wildcard-import
+from CommonServerUserPython import *  # noqa
```
Comment (bakatzir) [inline Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.py:3]:
```suggestion
from CommonServerPython import *
from CommonServerUserPython import *
```

File: Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.py [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+        )
+        return [relationships_lst.to_indicator()]
+
+
+def main() -> None:
+    params = {k: v for k, v in demisto.params().items() if v is not None}
+    params['indicator_type'] = FeedIndicatorType.File
+    params['feed_name_to_config'] = {
+        'File': {
+            'url': params.get("url") + "/api/v1/",
```
Comment (bakatzir) [inline Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.py:27]:
```suggestion
            'url': f'{params.get("url")}/api/v1/',
```

File: Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.py [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+                'reporter': 'reported_by',
+                'signature': 'malware_family',
+                'imphash': 'imphash',
+                'ssdeep': 'ssdeep',
+                'tags': 'tags'
+            }
+        }
+    }
+    params['data'] = {'query': 'get_recent',
+                      'selector': 'time'}
```
Comment (bakatzir) [inline Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.py:54]:
```suggestion
    params['data'] = {'query': 'get_recent',
                      'selector': 'time'} # query params to get only the recent changes for the incremental feed
```

**Review by bakatzir — COMMENTED** [review]
nice work.
please:
1. contact eynav refael for the image.
2. add the test playbook

File: Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed_description.md (hunk 4 above)
Comment (bakatzir) [reply Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed_description.md:1]: e.g:
https://github.com/demisto/content/blob/master/Packs/Zimperium/Integrations/Zimperium/Zimperium_description.md
https://github.com/demisto/content/blob/master/Packs/QRadar/Integrations/QRadar_v2/QRadar_v2_description.md#query-to-fetch-offenses

**Comment by ShirleyDenkberg** [comment]
@bziser In the Integration README it says: "This integration was integrated and tested with version **xx** of MalwareBazaar Feed."
Please update with the correct version of the feed.
I did not review the description file since there is a comment to update the content.
In addition, there was no content in the pack README file.

**Comment by ShirleyDenkberg** [comment]
@yuvalbenshalom @bakatzir Doc review is complete except for the comments above directed to @bziser.

**Comment by bziser** [comment]
> @bziser In the Integration README it says: "This integration was integrated and tested with version **xx** of MalwareBazaar Feed."
> Please update with the correct version of the feed.
> I did not review the description file since there is a comment to update the content.
> In addition, there was no content in the pack README file.

Hi, there is no version on the feed site.

File: Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/README.md [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+***
+Gets indicators from the feed.
+
+
+#### Base Command
+
+`malwarebazzar-get-indicators`
+#### Input
+
+There are no input arguments for this command.
```
Comment (bakatzir) [inline Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/README.md:41]: there is an input, limit

File: Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/README.md [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+    | --- | --- | --- |
+    | Server URL |  | True |
+    | Fetch indicators |  | False |
+    | Indicator Reputation | Indicators from this integration instance will be marked with this reputation | False |
+    | Source Reliability | Reliability of the source providing the intelligence data | True |
+    | Traffic Light Protocol Color | The Traffic Light Protocol \(TLP\) designation to apply to indicators fetched from the feed | False |
+    |  |  | False |
+    |  |  | False |
+    | Feed Fetch Interval |  | False |
+    | Incremental Feed | Incremental feeds pull only new or modified indicators that have been sent from the integration. The determination if the indicator is new or modified happens on the 3rd-party vendor's side, and only indicators that are new or modified are sent to Cortex XSOAR. Therefore, all indicators coming from these feeds are labeled new or modified. | False |
```
Comment (bakatzir) [inline Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/README.md:20]: let's just rm it.

**Review by bakatzir — COMMENTED** [review]
nice.
I think the mapping for the download_url is missing.

File: Packs/MalwareBazaar/pack_metadata.json [hunk 16]
```diff
@@ -0,0 +1,17 @@
+{
+    "name": "MalwareBazaar Feed",
```
Comment (bakatzir) [inline Packs/MalwareBazaar/pack_metadata.json:2]:
```suggestion
    "name": "Malware Bazaar",
```

File: Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.py [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+            'relation_name': EntityRelationship.Relationships.INDICATOR_OF,
+            'reverse_relationship_name': EntityRelationship.Relationships.INDICATED_BY,
+            'relation_entity_b': 'signature',
+            'relation_entity_b_type': 'Malware',
+            'create_relations_function': custom_build_relationships,
+            'mapping': {
+                'sha256_hash': 'sha256',
+                'sha1_hash': 'sha1',
+                'md5_hash': 'md5',
+                'first_seen': 'firstseenbyfeed',
```
Comment (bakatzir) [inline Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.py:41]: should be `first_seen_by_source` IIRC

File: Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.py [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+                'last_seen': 'lastseenbyfeed',
```
Comment (bakatzir) [inline Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.py:42]: should be `last_seen_by_source` IIRC

File: Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.py [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+                'file_name': 'Associated File Names',
+                'file_size': 'size',
+                'file_type': 'filetype',
+                'reporter': 'reported_by',
+                'signature': 'malware_family',
```
Comment (bakatzir) [inline Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.py:47]: rm
```suggestion
```

File: Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.yml [hunk 20]
```diff
@@ -0,0 +1,127 @@
...
+  name: feedReputation
+  options:
+  - None
+  - Good
+  - Suspicious
+  - Bad
+  required: false
+  type: 18
+- additionalinfo: Reliability of the source providing the intelligence data.
+  defaultvalue: A - Completely reliable
```
Comment (bakatzir) [inline Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.yml:29]:
```suggestion
  defaultvalue: B - Usually reliable
```
```suggestion
  defaultvalue: A - Completely reliable
```

**Review by bakatzir — COMMENTED** [review]
open points:
csv download of 48 hours for first time
add url_download indicator field

File: Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.yml (hunk 20 above)
Comment (bziser) [reply Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.yml:29]: So A or B?

Comment (bakatzir) [reply Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.yml:29]: B :)

File: Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/README.md [hunk 21]
```diff
@@ -0,0 +1,50 @@
...
+| limit | The maximum number of results to return. The default value is 50. | Optional | 
+
+#### Context Output
+
+There is no context output for this command.
+
+#### Command Example
+``` ```
+
+#### Human Readable Output
```
Comment (bakatzir) [inline Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/README.md:48]: there is no example in the readme. please make sure to run the `demisto-sdk generate-docs -i <yml_path> -e <examples_path>` to generate a readme with actual results for the commands.

File: Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.py [hunk 22]
```diff
@@ -0,0 +1,58 @@
...
+            'url': f'{params.get("url")}/api/v1/',
+            'extractor': "data",
+            'indicator': 'sha256_hash',
+            'indicator_type': FeedIndicatorType.File,
+            'relation_name': EntityRelationship.Relationships.INDICATOR_OF,
+            'reverse_relationship_name': EntityRelationship.Relationships.INDICATED_BY,
+            'relation_entity_b': 'signature',
+            'relation_entity_b_type': 'Malware',
+            'create_relations_function': custom_build_relationships,
+            'mapping': {
```
Comment (bakatzir) [inline Packs/MalwareBazaar/Integrations/MalwareBazaarFeed/MalwareBazaarFeed.py:36]: still missing the url_download mapping. as we discussed, will probably need to be via a mapping function.

**Review by bakatzir — APPROVED** [review]

**Pull request merged**